### PR TITLE
Add Settings Update db functionnality and minor fixes/improvements:

### PR DESCRIPTION
**Pull request by Milerius**
- upgrade C++17 to C++20
- Remove manual linking to filesystem library
- Add unit tests
- Add update_setting in db
- Update update_setting in service
- Update documentation according to discussions

### CODE REVIEW

File: API_doc/Communication.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Each response contain at least **REQUEST_STATE** (see below).
 |*CONFIG_LOAD*| Load an existing config. |**CONFIG_KEY** *or* **READONLY_CONFIG_KEY**| **CONFIG_NAME**<br>**CONFIG_ID**|
 |*CONFIG_UNLOAD*| Unload config. |**CONFIG_ID**|*none*|
 |*CONFIG_INCLUDE*| Include a config in another |**CONFIG_ID**<br>**SRC**<br>**DEST**|*none*|
-|*SETTING_UPDATE*| Update or create setting |**CONFIG_ID**<br>**SETTING_NAME**<br>**SETTING_VALUE**|*none*|
+|*SETTING_UPDATE*| Update or create setting |**CONFIG_ID**<br>**SETTINGS_TO_UPDATE**|*none*|
```
Comment (Gaspard--) [inline API_doc/Communication.md]: Unclear what this means in the context of the file. Please add details above / below.

File: API_doc/json_recipes/setting_update.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
   "REQUEST_NAME": "SETTING_UPDATE",
   "CONFIG_ID": 42,
-  "SETTING_NAME": "foo",
-  "SETTING_VALUE": "bar"
+  "SETTINGS_TO_UPDATE": {
```
Comment (Gaspard--) [inline API_doc/json_recipes/setting_update.json]: Is this supposed to be a list? A json array might make more sens in that case.

File: CMakeLists.txt [hunk 3]
```diff
@@ -27,9 +27,8 @@ file(GLOB_RECURSE SOURCES_LIB lib/*.cpp)
 add_subdirectory(vendor/uvw)
 
 add_executable(service ${SOURCES_SERVICE})
-find_library(filesystem_lib NAMES stdc++fs)
 find_library(sqlite3_lib NAMES sqlite3)
-target_link_libraries(service PUBLIC uv_a ${filesystem_lib} ${sqlite3_lib})
+target_link_libraries(service PUBLIC uv_a ${sqlite3_lib})
```
Comment (Gaspard--) [inline CMakeLists.txt]: I'm not sure, but I think you may still need some of these options. TBH I wouldn't mind manual checks for the compiler version as a temporary fix.

File: service/db.hpp [hunk 4]
```diff
@@ -200,6 +231,38 @@ namespace raven
...
+    {
+        SUBCASE("setting update unknown id") {
+            config_db db{std::filesystem::current_path() / "albinos_service_test.db"};
+            setting_update setting_update_data{config_id_st{42}, {{"foo", "bar"}}};
+            CHECK_EQ(static_cast<short>(db.settings_update(setting_update_data)),
+                static_cast<short>(request_state::unknown_id));
+            std::filesystem::remove(std::filesystem::current_path() / "albinos_service_test.db");
+        }
+
+        SUBCASE("regular update setting") {
```
Comment (Gaspard--) [inline service/db.hpp]:
```suggestion
        SUBCASE("normal update setting") {
```
Unless you meant regular as in repeated every X.

File: service/db.hpp [hunk 5]
```diff
@@ -200,6 +231,38 @@ namespace raven
...
+            config_db db{std::filesystem::current_path() / "albinos_service_test.db"};
+            json::json config_create = R"({"REQUEST_NAME": "CONFIG_CREATE","CONFIG_NAME": "ma_config"})"_json;
+            auto config_create_answer = db.config_create(config_create);
+            setting_update setting_update_data{config_create_answer.config_id, {{"foo", "bar"}}};
+            CHECK_EQ(static_cast<short>(db.settings_update(setting_update_data)),
+                static_cast<short>(request_state::success));
+            std::filesystem::remove(std::filesystem::current_path() / "albinos_service_test.db");
+        }
+
+        SUBCASE("regular update multiple settings") {
```
Comment (Gaspard--) [inline service/db.hpp]:
```suggestion
        SUBCASE("normal update multiple settings") {
```
(see above)

File: tests/CMakeLists.txt [hunk 6]
```diff
@@ -1,6 +1,6 @@
 add_executable(service-test)
 target_sources(service-test PUBLIC service-test.cpp)
 target_include_directories(service-test PRIVATE ../vendor/doctest/doctest ../vendor/json/single_include/nlohmann ../service ../vendor/uvw/src ../vendor/uvw/deps/libuv/include ../vendor/strong_type/include/ ../vendor/expected ../vendor/sql/hdr)
-target_link_libraries(service-test ${filesystem_lib} uv_a ${sqlite3_lib})
+target_link_libraries(service-test uv_a ${sqlite3_lib})
```
Comment (Gaspard--) [inline tests/CMakeLists.txt]: see notes about other cmake

File: API_doc/json_recipes/setting_update.json (hunk 2 above)
Comment (Milerius) [reply API_doc/json_recipes/setting_update.json]: no this is an object

File: CMakeLists.txt (hunk 3 above)
Comment (Milerius) [reply CMakeLists.txt]: Seem's fix since gcc 8.2 !

File: API_doc/json_recipes/setting_update.json (hunk 2 above)
Comment (Gaspard--) [reply API_doc/json_recipes/setting_update.json]: I guess since it's key-value pairs, that makes sens.

File: tests/CMakeLists.txt (hunk 6 above)
Comment (Gaspard--) [reply tests/CMakeLists.txt]: Resolved above.

**Pull request merged**